StateOfChargeManager.h

```diff
@@ -0,0 +1,38 @@
+#ifndef STATEOFCHARGEMANAGER
+#define STATEOFCHARGEMANAGER
+
+class StateOfChargeManager : public BatteryManagementSystem
+{
+    public:
+    bool checkOptimumLimit(float stateOfCharge)
+    {
+        bool returnStatus = true;
+        if(stateOfCharge < MINIMUMSTATEOFCHARGE || stateOfCharge > MAXIMUMSTATEOFCHARGE)
+        {
+            displayStateOfChargeAlert();
+            returnStatus = false;
+        }
+        return returnStatus;
+    }
+    
+    Range getRange(float stateOfCharge)

```

Comment:
Always, while using a word like `get`, see if you can do better.
In this case, `classifyRange` would be better.